### PR TITLE
Print info for flags need migration in separate info box

### PR DESCRIPTION
**Pull request by meteorcloudy**
Example look: https://buildkite.com/bazel/pcloudy-test/builds/35

Fixes: https://github.com/bazelbuild/continuous-integration/issues/586

@laszlocsomor

### CODE REVIEW

**Comment by laszlocsomor** [comment]
I can't view https://buildkite.com/bazel/pcloudy-test/builds/35

Can you attach a screenshot?

**Comment by meteorcloudy** [comment]
Now you can ;)

File: buildkite/aggregate_incompatible_flags_test_result.py [hunk 1]
```diff
@@ -127,14 +127,16 @@ def print_projects_need_to_migrate(failed_jobs_per_flag):
...
+    for flag, jobs in reversed(list(failed_jobs_per_flag.items())):
         if jobs:
             github_url = INCOMPATIBLE_FLAGS[flag]
+            info_text = []
             info_text.append(f"* **{flag}** " + get_html_link_text(":github:", github_url))
             info_text += merge_and_format_jobs(jobs.values(), "  - **{}**: {}")
+            print_info(flag, "error", info_text)
     if len(info_text) == 1:
         return
+    info_text = ["#### Downstream projects need to migrate for the following flags"]
```
Comment (fweikert) [inline buildkite/aggregate_incompatible_flags_test_result.py]: Nit: add a colon at the end of the text

File: buildkite/aggregate_incompatible_flags_test_result.py [hunk 2]
```diff
@@ -127,14 +127,16 @@ def print_projects_need_to_migrate(failed_jobs_per_flag):
 
 
 def print_flags_need_to_migrate(failed_jobs_per_flag):
-    info_text = ["#### Downstream projects need to migrate for the following flags"]
-    for flag, jobs in failed_jobs_per_flag.items():
+    for flag, jobs in reversed(list(failed_jobs_per_flag.items())):
```
Comment (laszlocsomor) [inline buildkite/aggregate_incompatible_flags_test_result.py]: Can you comment and explain the need for `reversed`?

File: buildkite/aggregate_incompatible_flags_test_result.py [hunk 3]
```diff
@@ -127,14 +127,16 @@ def print_projects_need_to_migrate(failed_jobs_per_flag):
...
         if jobs:
             github_url = INCOMPATIBLE_FLAGS[flag]
+            info_text = []
             info_text.append(f"* **{flag}** " + get_html_link_text(":github:", github_url))
             info_text += merge_and_format_jobs(jobs.values(), "  - **{}**: {}")
+            print_info(flag, "error", info_text)
```
Comment (laszlocsomor) [inline buildkite/aggregate_incompatible_flags_test_result.py]: IIUC each flag gets a separate info box. Correct? Please add comment.

File: buildkite/aggregate_incompatible_flags_test_result.py (hunk 2 above)
Comment (fweikert) [inline buildkite/aggregate_incompatible_flags_test_result.py]: Why reversed()? Also, we need to sort the flags first.

Comment (meteorcloudy) [reply buildkite/aggregate_incompatible_flags_test_result.py]: Done.

File: buildkite/aggregate_incompatible_flags_test_result.py (hunk 3 above)
Comment (meteorcloudy) [reply buildkite/aggregate_incompatible_flags_test_result.py]: Done

File: buildkite/aggregate_incompatible_flags_test_result.py (hunk 1 above)
Comment (meteorcloudy) [reply buildkite/aggregate_incompatible_flags_test_result.py]: Done

File: buildkite/aggregate_incompatible_flags_test_result.py (hunk 2 above)
Comment (meteorcloudy) [reply buildkite/aggregate_incompatible_flags_test_result.py]: I added a comment about how using reverse. And good catch, I thought the flags were sorted.

**Comment by meteorcloudy** [comment]
Tested at https://buildkite.com/bazel/pcloudy-test/builds/37#61db0522-ba71-42ab-bc8e-13309de4efd4

**Pull request merged**